esic_fe/src/components/Navbar/Navbar.jsx

```diff
@@ -1,12 +1,57 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
 
 export function Navbar() {
-    return (
-        <nav className="text-secondary-900">
-        <ul>
-            Navbar
-        </ul>
+    
+    const [navbar,setnavbar] = useState(false);
+    const [menu,setmenu] = useState(false)
+    
+    const showmenu = () =>{
+        setmenu((prev) => !prev)
+    }
+    const changebg = () =>{
+        if (window.scrollY >= 100){
+            setnavbar(true)
+        } else{
+            setnavbar(false)
+        }
+    }
+    window.addEventListener('scroll',changebg);
```

Comment:
Adding event listeners directly inside the component body can lead to performance issues. It's better to handle this within a `useEffect` hook.

```suggestion
    useEffect(() => {
    window.addEventListener('scroll', changeBg);
    return () => {
        window.removeEventListener('scroll', changeBg);
    };
}, []);
```

lmk if you face any issues changing this.

Also a suggestion would be to have a bg color from the start itself

Reply:
Okay, so do we need a single bg color throughout or it should change with scroll?

Reply:
Keep it single bg throughout

Reply:
I put this sinnpet in ChatGPT just for an explaination there it used a term _**component mounts**_, so here component is `useEffect`, mounts as soon as application loads and unmounts as we close it, right?

Reply:
> I put this sinnpet in ChatGPT just for an explaination there it used a term _**component mounts**_, so here component is `useEffect`, mounts as soon as application loads and unmounts as we close it, right?

**component mounts** is when the component is first added to the DOM

**component unmounts** is when the component is removed from the DOM

`useEffect` runs after the component mounts and cleans up before the component unmounts.

to clear out - `useEffect` itself is not the component, but a hook that runs side effects (like adding event listeners) when the component mounts and cleans up those side effects when it unmounts.

Reply:
okay